c2corg_ui/static/js/elevationprofile.js

```diff
@@ -0,0 +1,443 @@
+    restrict: 'E',
+    controller: 'appElevationProfileController',
+    controllerAs: 'elevationProfileCtrl',
+    templateUrl: '/static/partials/elevationprofile.html'
+  };
+};
+app.module.directive('appElevationProfile', app.elevationProfileDirective);
+
+/**
+ * @param {angular.Scope} $scope Scope.
+ * @param {?GeoJSONFeatureCollection} mapFeatureCollection FeatureCollection of
+ *    features shown on the map.
+ * @param {app.Lang} appLang Lang service.
+ * @param {ngeo.Debounce} ngeoDebounce ngeo Debounce service.
+ * @constructor
+ * @ngInject
+ */
+app.ElevationProfileController = function(
+  $scope,
+  mapFeatureCollection,
+  appLang,
+  ngeoDebounce
+) {
+  /**
+   * @type {angular.Scope}
+   * @private
+   */
+  this.scope_ = $scope;
+
+  /**
+   * @type {?GeoJSONFeatureCollection}
+   * @private
+   */
+  this.featureCollection_ = mapFeatureCollection;
+
+  /**
+   * @private
+   * @type {ngeo.Debounce}
+   */
+  this.ngeoDebounce = ngeoDebounce;
+
+  /**
+   * @export
+   */
+  this.mode = 'distance';
+
+  this.i18n = {
+    elevation: appLang.translate('Elevation'),
+    elevation_legend: appLang.translate('Elevation (m)'),
+    meters: appLang.translate('meters'),
+    distance: appLang.translate('Distance'),
+    distance_legend: appLang.translate('Distance (km)'),
+    km: appLang.translate('kilometers'),
+    time: appLang.translate('Time'),
+    duration: appLang.translate('Duration'),
+    duration_legend: appLang.translate('Duration (hrs)')
+  };
+
+  var coords = mapFeatureCollection.features.filter(function(feature) {
+    return feature.geometry.type === 'LineString';
+  })[0].geometry.coordinates;
+
+  // we consider the track contains time if all points have this information
+  var timeAvailable = coords.every(function(coord) {
+    return coords.length > 3;
+  });
+  this.timeAvailable = timeAvailable;
+  var startDate = timeAvailable ? new Date(coords[0][3] * 1000) : undefined;
+  var totalDist = 0;
+
+  this.data = coords.map(function(coord, i, coords) {
+    var date = timeAvailable ? new Date(coord[3] * 1000) : undefined;
+    var d = 0;
+    if (i > 0) {
+      // convert from web mercator to lng/lat
+      var deg1 = ol.proj.transform(
+        [coords[i][0], coords[i][1]],
+        'EPSG:3857',
+        'EPSG:4326'
+      );
+      var deg2 = ol.proj.transform(
+        [coords[i - 1][0], coords[i - 1][1]],
+        'EPSG:3857',
+        'EPSG:4326'
+      );
+      // arc distance x earth radius
+      d = window.d3.geo.distance(deg1, deg2) * 6371;
+    }
+    totalDist += d;
+    return {
+      date: date,
+      ele: coord[2] || 0,
+      d: totalDist,
+      elapsed: timeAvailable ? date - startDate : undefined
+    };
+  });
+  if (
+    !this.data.find(function(coord) {
+      return coord.ele > 0;
+    })
+  ) {
+    $('#elevation-profile').closest('.view-details-info').remove();
+    return;
+  }
+  this.createChart();
+};
+
+app.ElevationProfileController.prototype.createChart = function() {
```

Comment:
Perhaps you could add the usual "export" keyword (ditto for the other functions):
```
/**
 * @export
 */
```